bpf/process/types/basic.h

```diff
@@ -1211,6 +1211,18 @@ filter_64ty_selector_val(struct selector_arg_filter *filter, char *args)
 					return 1;
 			}
 			break;
+		case op_filter_capability: {
+			__u64 caps;
+
+			// We resolve the kernel_cap_t/cap_* here
+			probe_read(&caps, sizeof(__u64), args);
```

Comment:
I still think this should work without probe_read, like that this should be enough:

```
+               case op_filter_capability: {
+                       if (*(u64 *)args & w)
+                               return 1;
+                       break;
+               }
```

so it could share the code with op_filter_mask and maybe 4.19 would work?

could you please explain in more details what's the idea behind the resolve?

Reply:
You are actually very right. I was pretty much sure that it was needed, but it works without it. I'll do the change, thank you 💯

Reply:
Done, thank you